### PR TITLE
fix for states issues

### PR DESCRIPTION
**Pull request by elopezanaya**
### Description
_Include a description of the problem to be solved_

unstable states  for OOH widgets, showing for a brief milisecond a blue button , then re-renders as OOH , this causes when clicking the button to start a chat, since the on click  function is overrided , or buttons are clashing in the same space

## Solution Proposed
_Detail what is the solution proposed, include links to design document if required or any other document required to support the solution_

Adding fixes to handle states properly maintaining validation for close conversation

### Acceptance criteria
_Define what are the conditions to consider the PR has achieved the intended goal_

## Test cases and evidence
_Include what tests cases were considered, any evidence of testing for future references, to identify any corner cases, etc_
Video Project 17
Video Project 18

## A11y
- [ ] You 
have run the [Automated Accessibility Check](https://accessibilityinsights.io/docs/en/windows/getstarted/automatedchecks) and have no reported issues

__*Please provide justification if any of the validations has been skipped.*__

### CODE REVIEW

File: chat-widget/src/components/chatbuttonstateful/ChatButtonStateful.tsx [hunk 1]
```diff
@@ -111,8 +121,13 @@ export const ChatButtonStateful = (props: IChatButtonStatefulParams) => {
...
             const isOutsideOperatingHours = state.appStates.outsideOperatingHours;
             setOutOfOperatingHours(isOutsideOperatingHours);
+        } else {
+            // If conversation state is not Closed, we should not be in out-of-office mode
+            if (outOfOperatingHours) {
+                setOutOfOperatingHours(false);
+            }
         }     
-    }, [state.appStates.conversationState]);
+    }, [state.appStates.conversationState, state.appStates.outsideOperatingHours, state.appStates.isMinimized]);
```
[the remark this answers is not in the document]
Comment (elopezanaya) [reply chat-widget/src/components/chatbuttonstateful/ChatButtonStateful.tsx:130]: outOfOperationhours takes its value from state.appStates.outsideOperatingHours, no need to pass it in the dependencies

**Review by mithunjohny — APPROVED** [review]
